app/com/arpnetworking/metrics/portal/hosts/impl/MultiProvider.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2016 Smartsheet.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.hosts.impl;
+
+import akka.actor.Actor;
+import akka.actor.Props;
+import akka.actor.UntypedActor;
+import com.arpnetworking.play.configuration.ConfigurationHelper;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.google.inject.Inject;
+import com.google.inject.assistedinject.Assisted;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValueType;
+import play.Configuration;
+import play.Environment;
+
+import java.util.Set;
+
+/**
+ * Provider that starts multiple sub providers.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public class MultiProvider extends UntypedActor {
+    /**
+     * Public constructor.
+     *
+     * @param factory a {@link HostProviderFactory}.
+     * @param environment Play environment.
+     * @param configuration Play configuration.
+     */
+    @Inject
+    public MultiProvider(final HostProviderFactory factory, final Environment environment, @Assisted final Configuration configuration) {
+        final Set<String> entries = configuration.subKeys();
+        final Config underlying = configuration.underlying();
+        for (String key : entries) {
+            if (underlying.getValue(key).valueType() == ConfigValueType.OBJECT) {
+                if (!underlying.hasPath(key + ".type")) {
+                    LOGGER.warn()
+                            .setMessage("Expected type for host provider")
+                            .addData("key", key)
+                            .log();
+                }
+                final Configuration subConfig = configuration.getConfig(key).withFallback(configuration);
```

Comment:
This fallback should never be executed since you are iterating the subkeys -- correct? If this does fail (for some other reason) proceeding with the configuration will likely produce some strange results.